src/mpid/ch4/netmod/ucx/ucx_am.h

```diff
@@ -17,7 +17,11 @@ MPL_STATIC_INLINE_PREFIX void MPIDI_UCX_am_isend_callback(void *request, ucs_sta
     MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDI_UCX_AM_ISEND_CALLBACK);
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDI_UCX_AM_ISEND_CALLBACK);
 
-    MPIR_gpu_free_host(req->dev.ch4.am.netmod_am.ucx.pack_buffer);
+    if (req->dev.ch4.am.netmod_am.ucx.is_gpu) {
+        MPIR_gpu_free_host(req->dev.ch4.am.netmod_am.ucx.pack_buffer);
+    } else {
+        MPL_free(req->dev.ch4.am.netmod_am.ucx.pack_buffer);
+    }
```

Comment:
I am a bit confused. Shouldn't we always use `MPL_malloc` and `MPL_free`?

Reply:
Oh, I guess I assumed that registered buffers were desirable in some cases. If the takeaway is that malloc always performs better, I will just update them.

Reply:
That's my take away from https://github.com/pmodels/mpich/issues/5171#issue-839174760.

Let's keep the changes minimal for now. I believe @yfguo is working on adding genq to the ucx am path, so these code will be replaced anyway.